app/src/main/java/de/danoeh/antennapod/dialog/IntraFeedSortDialog.java

```diff
@@ -16,35 +16,63 @@
     @NonNull
     protected Context context;
 
+    private String[] localSortItems;
+    private SortOrder[] localSortValues;
+
+    private String[] commonSortItems;
+    private SortOrder[] commonSortValues;
+
     public IntraFeedSortDialog(@NonNull Context context, @Nullable SortOrder sortOrder) {
         this.context = context;
         this.currentSortOrder = sortOrder;
+        setSortOptions();
     }
 
-    public void openDialog() {
-        final String[] items = context.getResources().getStringArray(R.array.feed_episodes_sort_options);
-        final String[] valueStrs = context.getResources().getStringArray(R.array.feed_episodes_sort_values);
-        final SortOrder[] values = new SortOrder[valueStrs.length];
-        for (int i = 0; i < valueStrs.length; i++) {
-            values[i] = SortOrder.valueOf(valueStrs[i]);
-        }
+    private void setSortOptions() {
+        localSortItems = context.getResources().getStringArray(R.array.local_feed_episodes_sort_options);
+        final String[] localSortStringValues = context.getResources().getStringArray(R.array.local_feed_episodes_sort_values);
+        localSortValues = SortOrder.getSortOrderValuesFromStringValues(localSortStringValues);
+
+        commonSortItems = context.getResources().getStringArray(R.array.feed_episodes_sort_options);
+        final String[] commonSortStringValues = context.getResources().getStringArray(R.array.feed_episodes_sort_values);
+        commonSortValues = SortOrder.getSortOrderValuesFromStringValues(commonSortStringValues);
+
+    }
 
-        int idxCurrentSort = 0;
-        for  (int i = 0; i < values.length; i++) {
+    public void openDialog(boolean showSortOptionsLocalFeed) {
```

Comment:
If you move the argument to the constructor instead, you only have to load one of the arrays. Currently both are read and only one is actually used.